### PR TITLE
[X86][AMX] Add AMX FP8 new APIs

### PR DESCRIPTION
**Pull request by fzou1**
This is a follow-up to #113850.

Ref.: https://cdrdv2.intel.com/v1/dl/getContent/671368

### CODE REVIEW

File: clang/lib/Headers/amxfp8intrin.h [hunk 1]
```diff
@@ -15,81 +15,214 @@
...
+}
+
+#define _tile_dpbf8ps(dst, src1, src2)                                         \
+  __builtin_ia32_tdpbf8ps((dst), (src1), (src2))
+#define _tile_dpbhf8ps(dst, src1, src2)                                        \
+  __builtin_ia32_tdpbhf8ps((dst), (src1), (src2))
+#define _tile_dphbf8ps(dst, src1, src2)                                        \
+  __builtin_ia32_tdphbf8ps((dst), (src1), (src2))
+#define _tile_dphf8ps(dst, src1, src2)                                         \
+  __builtin_ia32_tdphf8ps((dst), (src1), (src2))
```
Comment (phoebewang) [inline clang/lib/Headers/amxfp8intrin.h:226]: #undef __DEFAULT_FN_ATTRS_FP8

**Comment by phoebewang** [comment]
Missing IR test?

File: clang/lib/Headers/amxfp8intrin.h (hunk 1 above)
Comment (fzou1) [reply clang/lib/Headers/amxfp8intrin.h:226]: Done.

**Comment by fzou1** [comment]
> Missing IR test?

Sorry. Added. Thanks.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**